Docs/src/tutorials/basictutorials/basictutorial2.md

```diff
@@ -0,0 +1,197 @@
+
+@note You must be wondering why we call setDirection for light source and then call the same method for related SceneNode. As long as setDirection is depricated for Light it has to be attached to SceneNode. Since SceneNode default rotation is NEGATIVE\_UNIT\_Z we need to set same value for Light. Direction for light by default is not NEGATIVE\_UNIT\_Z but in the future releases it will be replaced with this value. So this extra call of setDirection for Light could be deleted in future. In other words we need this extra move for future compatibilty.
+
+![](bt2_light_dir_1.png)
+
+Finally, we set what is called the spotlight range. These are the angles that determine where the light fades from bright in the middle to dimmer on the outside edges.
+
+@snippet Samples/Tutorials/BasicTutorial2.cpp spotlightrange
+Compile and run the application. You should see the shadowy blue figure of a ninja.
+
+![](bt2_ninja1.jpg)
+
+# Creating More Lights {#CreatingMoreLights}
+Next we'll add a directional light to our scene. This type of light essentially simulates daylight or moonlight. The light is cast at the same angle across the entire scene equally. As before, we'll start by creating the Light and setting its type.
+
+@snippet Samples/Tutorials/BasicTutorial2.cpp directlight
+
+Now we'll set the diffuse and specular colors to a dark red.
+@snippet Samples/Tutorials/BasicTutorial2.cpp directlightcolor
+
+Finally, we need to set the Light's direction. A directional light does not have a position because it is modeled as a point light that is infinitely far away.
+
+@snippet Samples/Tutorials/BasicTutorial2.cpp directlightdir
+
+![](bt2_light_dir_2.png)
+
+The Light class also defines a Ogre::Light::setAttenuation function which allows you to control how the light dissipates as you get farther away from it. After you finish this tutorial, try using this method in your scene to see how it affects your lights.
+
+Compile and run the application. Your ninja should now have a shadow cast behind him, and the scene should be filled with red light.
+
+![](bt2_ninja2.jpg)
+
+To complete the set, we will now add a point light to our scene.
+@snippet Samples/Tutorials/BasicTutorial2.cpp pointlight
+
+We'll set the the specular and diffuse colors to a dark gray.
+
+@snippet Samples/Tutorials/BasicTutorial2.cpp pointlightcolor
+
+A point light has no direction. It only has a position. We will place our last light above and behind the ninja.
+
+@snippet Samples/Tutorials/BasicTutorial2.cpp pointlightpos
+
+Compile and run the application. You should see a long shadow cast in front of the ninja now. And you should see the effects of the point light brightening up the area behind the ninja. Try to think about why the colors turn out the way they do. For instance, why does the shadow behind the ninja appear to have no red at all?
+
+![](bt2_ninja3.jpg)
+
+# Shadow Types {#ShadowTypes}
+Ogre currently supports three types of Shadows:
```

Comment:
actually it is six. probably it is better to just reference the enum here: https://ogrecave.github.io/ogre/api/1.10/group___general.html#ga79dcd426d291c31072c1ad6f183715d6